compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseEncodingOpsIntoDispatchRegions.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/Encoding/IR/EncodingDialect.h"
+#include "iree/compiler/Dialect/Encoding/IR/EncodingOps.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.h"
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Interfaces/FunctionInterfaces.h"
+
+#define DEBUG_TYPE "iree-flow-clone-producers-into-dispatch-regions"
+
+namespace mlir::iree_compiler::IREE::Flow {
+
+#define GEN_PASS_DEF_FUSEENCODINGOPSINTODISPATCHREGIONSPASS
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h.inc"
+
+namespace {
+
+struct FuseEncodingOpsIntoDispatchRegionsPass
+    : public IREE::Flow::impl::FuseEncodingOpsIntoDispatchRegionsPassBase<
+          FuseEncodingOpsIntoDispatchRegionsPass> {
+  void runOnOperation() override {
+    mlir::FunctionOpInterface funcOp = getOperation();
+    IRRewriter rewriter(funcOp->getContext());
+
+    FormDispatchRegionsPassOptions options;
+    funcOp->walk([&](IREE::Encoding::SetEncodingOp encodingOp) {
+      OpOperand &operand = encodingOp->getOpOperand(0);
```

Comment:
I think what you're looking for is `encodingOp.getSourceMutable()`. It avoids the magic number `0`.